### PR TITLE
Allow disabling tests that require native clang

### PR DESCRIPTION
**Pull request by dschuff**
Add EMTEST_NO_NATIVE_CLANG and a decorator that can apply to test functions.

### CODE REVIEW

File: tests/runner.py [hunk 1]
```diff
@@ -100,6 +100,8 @@ def delete_contents(pathname):
 
 EMTEST_SKIP_SLOW = os.getenv('EMTEST_SKIP_SLOW')
 
+EMTEST_NO_NATIVE_CLANG = os.getenv('EMTEST_NO_NATIVE_CLANG')
```
Comment (kripken) [inline tests/runner.py:103]: we have `EMTEST_LACKS_SOUND_HARDWARE` and a few others in browser, so perhaps this could be `EMTEST_LACKS_NATIVE_CLANG`

**Comment by dschuff** [comment]
I assume you mean clang_native.py? 😉 
What would we assert on?
There are several places now where we raise exceptions if there is no SDK found. On the bot we were hitting an unbound local error where we should have been raising one of those exceptions (that's fixed in this PR).

**Comment by dschuff** [comment]
let's see if we can unblock the roll

**Comment by sbc100** [comment]
Calling any function in tests/clang_native.py could assert if EMTEST_LACKS_NATIVE_CLANG.

That would means that a test should have been skipped but wasn't.

**Pull request merged**